### PR TITLE
Hanaol/dataloader 2

### PR DESCRIPTION
**Pull request by hanaol**
**Problem**

The current data loader has the following issues:

- It relies on separate scripts within the data registry for handling the MP and QM9 datasets, even though they share common utilities.

- It performs the train–validation split on the fly, without explicit support for a held-out test set.

**Solution**

- The data loader has been refactored to address these issues.

- The data module and its configuration are now defined in the configuration files and loaded with `Hydra`, reducing data-loader boilerplate and improving modularity and reproducibility.

Note: The root key in the configuration should point to a directory with the following structure: 
```
root/
├── mp_filelist.txt   # List of sample indices
├── split.json        # Dictionary with 'train', 'validation', and 'test' keys,
│                     # each mapped to a list of indices referring to mp_filelist.txt or qm9_filelist.txt
├── data/             
└── label/         
```

### CODE REVIEW

**Comment by hanaol** [comment]
@forklady42 This is a reminder to discuss dataloader `shuffle `and whether we need to explicitly set a `DistributedSampler`.

File: src/electrai/dataloader/dataset.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        return DataLoader(
+            self.subsets["validation"],
+            self.batch_size,
+            num_workers=self.val_workers,
+            shuffle=False,  # I added this
+            # collate_fn=partial(collate_list_of_dicts, pin_memory=self.pin_memory),
+            # note: no sampler, so all devices will get full set
+        )
+
+    def test_dataloader(self):
```
Comment (forklady42) [inline src/electrai/dataloader/dataset.py]: This doesn't seem to be used anywhere and would raise a `KeyError` because `test` is not included in subsets.

File: src/electrai/dataloader/dataset.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+from src.electrai.dataloader.split import split_data
+from torch.utils.data import DataLoader, Dataset
+
+if TYPE_CHECKING:
+    import os
+
+dtype_map = {"f32": torch.float32, "f16": torch.float16, "bf16": torch.bfloat16}
+
+
+class RhoRead:
```
Comment (forklady42) [inline src/electrai/dataloader/dataset.py]: Given the Lightning refactor, having `RhoRead` inherit from `LightningDataModule` would better integrate with the rest of the Lightning setup, and give access to Lightning's built-in automatic distributed sampling and checkpoint hooks.

**Review by hanaol — COMMENTED** [review]
addressed
